torch/distributed/_tensor/ops/tensor_ops.py

```diff
@@ -57,6 +46,22 @@ def default_strategy(mesh: DeviceMesh, op_schema: OpSchema) -> StrategyType:
     )
 
 
+register_op_strategy(
+    [
+        aten.clone.default,
+        aten.contiguous.default,
+        aten.copy_.default,
+        aten.detach.default,
+        aten.fill_.Scalar,
+        aten.zero_.default,
+    ]
+)(default_strategy)
+
+register_op_strategy(
+    aten._to_copy.default, schema_info=RuntimeSchemaInfo(static_kwargkey=["dtype"])
+)(default_strategy)
```

Comment:
Just wanna confirm my understanding of the fix:

It looks like `static_kwargkey` ends up getting used in the cache-lookup for DTensor's sharding prop: https://github.com/pytorch/pytorch/blob/main/torch/distributed/_tensor/op_schema.py#L299

So was the problem that we had a graph with `_to_copy()` showing up twice, with different dtypes, and we ended up using the cached sharding strategy, when we should have recomputed it for a different dtype?

Reply:
Yes! this is correct, we should recomputed the dtype in this case but it was reusing the cached type